client/src/composables/persistentToggle.ts

```diff
@@ -0,0 +1,28 @@
+import { type Ref, ref, watch } from "vue";
+
+interface ToggleStateInterface {
+    toggled: Ref<boolean>;
+    toggle: () => void;
+}
+
+export function usePersistentToggle(uniqueId: string): ToggleStateInterface {
+    const localStorageKey = `toggle-state-${uniqueId}`;
+
+    // Retrieve the toggled state from localStorage if available, otherwise default to false
+    const toggled = ref<boolean>(localStorage.getItem(localStorageKey) === "true");
+
+    // Watch for changes in the toggled state and persist them to localStorage
+    watch(toggled, (newVal: boolean) => {
+        localStorage.setItem(localStorageKey, String(newVal));
+    });
```

Comment:
```suggestion
import { type Ref } from "vue";

import { useUserLocalStorage } from "./userLocalStorage";

interface ToggleStateInterface {
    toggled: Ref<boolean>;
    toggle: () => void;
}

export function usePersistentToggle(uniqueId: string): ToggleStateInterface {
    const localStorageKey = `toggle-state-${uniqueId}`;
    const toggled = useUserLocalStorage(localStorageKey, true);
```

Reply:
Good call on reusing the localstorage composable